packages/renderer/src/lib/container/ContainerDetailsTerminal.spec.ts

```diff
@@ -112,3 +114,78 @@ test('expect being able to reconnect ', async () => {
   // creating a new terminal requires new shellInContainer call
   expect(shellInContainerMock).toHaveBeenCalledTimes(2);
 });
+
+test('terminal active/ restarts connection after stopping and starting a container', async () => {
+  const container: ContainerInfoUI = {
+    id: 'myContainer',
+    state: 'RUNNING',
+    engineId: 'podman',
+  } as unknown as ContainerInfoUI;
+
+  let onDataCallback: (data: Buffer) => void = () => {};
+
+  const sendCallbackId = 12345;
+  shellInContainerMock.mockImplementation(
+    (
+      _engineId: string,
+      _containerId: string,
+      onData: (data: Buffer) => void,
+      _onError: (error: string) => void,
+      _onEnd: () => void,
+    ) => {
+      onDataCallback = onData;
+      // return a callback id
+      return sendCallbackId;
+    },
+  );
+
+  // render the component with a terminal
+  const renderObject = render(ContainerDetailsTerminal, { container, screenReaderMode: true });
+
+  // wait shellInContainerMock is called
+  await waitFor(() => expect(shellInContainerMock).toHaveBeenCalled());
+
+  // write some data on the terminal
+  onDataCallback(Buffer.from('hello\nworld'));
+
+  // wait 1s
+  await new Promise(resolve => setTimeout(resolve, 1000));
+
+  // search a div having aria-live="assertive" attribute
+  const terminalLinesLiveRegion = renderObject.container.querySelector('div[aria-live="assertive"]');
+
+  // check the content
+  expect(terminalLinesLiveRegion).toHaveTextContent('hello world');
+
+  container.state = 'EXITED';
+  console.log(container);
+
+  await renderObject.rerender({ container: container, screenReaderMode: true });
+
+  await tick();
+
+  await new Promise(resolve => setTimeout(resolve, 100));
+
+  expect(screen.queryByText('Container is not running')).toBeInTheDocument();
```

Comment:
need to be in waitFor

also it's strange if tick is not doing the update ?
missing a new tick() ?